lua/telescope/previewers.lua

```diff
@@ -381,19 +381,56 @@ previewers.vim_buffer = defaulter(function(_)
   }
 end, {})
 
-previewers.git_commit_diff = defaulter(function(_)
+previewers.git_commit_diff_to_parent = defaulter(function(opts)
   return previewers.new_termopen_previewer {
     get_command = function(entry)
-      local sha = entry.value
-      return { 'git', '-p', 'diff', sha .. '^!' }
+      local command = { 'git', '--paginate', 'diff', entry.value .. '^!' }
+
+      if opts.relative_file_path ~= nil then
+        table.insert(command, '--')
+        table.insert(command, opts.relative_file_path)
+      end
```

Comment:
you can also do `table.insert(command, {'--', opts.current_file })` and then `return vim.tbl_flatten(command)`.

But i think that doesn't really matter so just for information sakes :)

Reply:
I appreciate the tips. I'm still not too familiar with Lua, so I'm glad to learn new tricks.